jobs/resource_job.go

```diff
@@ -572,16 +572,14 @@ func (a JobsAPI) Read(id string) (job Job, err error) {
 		job.Settings.sortWebhooksByID()
 	}
 
-	if job.RunAsUserName != "" && job.Settings != nil {
-		userNameIsEmail := strings.Contains(job.RunAsUserName, "@")
-
-		if userNameIsEmail {
+	if job.Settings != nil && job.RunAsUserName != "" && job.RunAsUserName != job.CreatorUserName {
```

Comment:
Why the addition of `job.RunAsUserName != job.CreatorUserName`? I see that it is happening but don't quite understand why.

Reply:
Right now `job.RunAsUserName` is populated every time, even without `run_as` specified. As a result, we generate not required blocks in the exported code.